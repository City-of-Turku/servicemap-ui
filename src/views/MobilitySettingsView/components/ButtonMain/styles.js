const styles = theme => ({
  button: {
    color: 'rgba(0, 0, 0, 255)',
    width: '100%',
    height: '3.125rem',
    background: 'rgba(245, 245, 245, 255)',
    textTransform: 'capitalize',
    justifyContent: 'flex-start',
    borderRadius: '0',
    borderTop: '1px solid rgba(0, 0, 0, 255)',
    borderBottom: 'none',
    '&:hover': {
      background: 'rgba(230, 230, 230, 255)',
    },
  },
  active: {
    borderBottom: '1px solid #6f7276',
    borderTop: '1px solid #6f7276',
    background: 'rgba(70, 72, 75, 255)',
    color: '#fff',
    '&:hover': {
      color: '#fff',
      background: '#3e3f42',
      borderBottom: '1px solid #6f7276',
      borderTop: '1px solid #6f7276',
    },
  },
  iconActive: {
    fill: '#fff',
    width: '40px',
    height: '40px',
    marginRight: theme.spacing(1),
  },
  icon: {
    fill: '#000',
    width: '40px',
    height: '40px',
    marginRight: theme.spacing(1),
  },
});

export default styles;
