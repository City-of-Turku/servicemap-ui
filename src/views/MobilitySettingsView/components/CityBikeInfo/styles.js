const styles = theme => ({
  container: {
    padding: theme.spacing(2),
    textAlign: 'left',
    paddingBottom: theme.spacing(1),
  },
  margin: {
    marginBottom: theme.spacing(1),
  },
  bold: {
    fontWeight: 'bold',
  },
});

export default styles;
