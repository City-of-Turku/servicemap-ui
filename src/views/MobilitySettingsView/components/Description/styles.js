const styles = theme => ({
  buttonWhite: {
    width: '100%',
    textAlign: 'left',
    textTransform: 'none',
    padding: theme.spacing(1),
  },
  paragraph: {
    textAlign: 'left',
    padding: theme.spacing(1.5),
  },
  subtitle: {
    textTransform: 'none',
  },
  toggleText: {
    width: '85%',
    textAlign: 'left',
  },
});

export default styles;
