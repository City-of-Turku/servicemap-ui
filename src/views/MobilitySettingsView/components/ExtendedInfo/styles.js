export default theme => ({
  container: {
    margin: theme.spacing(2),
    textAlign: 'left',
    paddingBottom: theme.spacing(2),
  },
  margin: {
    marginBottom: theme.spacing(1),
  },
});
