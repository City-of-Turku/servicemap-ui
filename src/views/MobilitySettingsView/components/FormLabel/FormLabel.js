import { FormControlLabel, Switch, Typography } from '@mui/material';
import useMediaQuery from '@mui/material/useMediaQuery';
import PropTypes from 'prop-types';
import React from 'react';

/**
 * Render 1 or more switches inside form.
 * @property {any} classes
 * @property {any} intl
 * @property {string} keyVal
 * @property {string} msgId
 * @property {boolean} checkedValue
 * @property {Function} onChangeValue
 * @return {JSX Element}
 */

const FormLabel = ({
  classes, intl, msgId, checkedValue, onChangeValue,
}) => {
  const useMobileStatus = () => useMediaQuery('(max-width:360px)');

  const isNarrow = useMobileStatus();

  return (
    <FormControlLabel
      label={(
        <Typography
          variant="body2"
          aria-label={intl.formatMessage({
            id: msgId,
          })}
        >
          {intl.formatMessage({
            id: msgId,
          })}
        </Typography>
      )}
      control={(
        <Switch
          checked={checkedValue}
          role="switch"
          inputProps={{
            'aria-label': intl.formatMessage({
              id: msgId,
            }),
          }}
          onChange={onChangeValue}
          onKeyPress={(event) => {
            if (event.key === 'Enter') {
              onChangeValue();
            }
          }}
        />
      )}
      className={`${classes.formLabel} ${isNarrow ? classes.paddingSm : classes.paddingMd}`}
    />
  );
};

FormLabel.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  msgId: PropTypes.string,
  checkedValue: PropTypes.bool,
  onChangeValue: PropTypes.func.isRequired,
};

FormLabel.defaultProps = {
  msgId: '',
  checkedValue: false,
};

export default FormLabel;
