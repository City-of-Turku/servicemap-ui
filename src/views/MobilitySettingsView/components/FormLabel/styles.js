const styles = {
  formLabel: {
    backgroundColor: 'rgba(70,72,75,255)',
    margin: '0',
    color: '#fff',
  },
  paddingMd: {
    padding: '0.5rem 3rem',
  },
  paddingSm: {
    padding: '0.5rem 0.8rem',
  },
};

export default styles;
