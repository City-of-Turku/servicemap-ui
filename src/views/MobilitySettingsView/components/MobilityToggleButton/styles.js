const styles = (theme) => ({
  mobilityMapSwitch: {
    paddingLeft: theme.spacing(2),
    display: 'inline-flex',
    alignItems: 'center',
    marginLeft: theme.spacing(2),
    verticalAlign: 'middle',
  },
  labelContainer: {
    marginLeft: theme.spacing(2),
  },
  customSwitch: {
    overflow: 'visible',
  },
  switchBorder: {
    border: '1px solid #949494',
  },
});

export default styles;
