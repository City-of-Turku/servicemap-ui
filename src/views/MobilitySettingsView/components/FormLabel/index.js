import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import FormLabel from './FormLabel';
import styles from './styles';

export default withStyles(styles)(injectIntl(FormLabel));
