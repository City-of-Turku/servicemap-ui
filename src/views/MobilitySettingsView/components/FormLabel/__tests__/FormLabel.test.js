// Link.react.test.js
import React from 'react';
import { ThemeProvider } from '@mui/material/styles';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../themes';
import FormLabel from '../index';
import finnishTranslations from '../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  msgId: 'mobilityPlatform.menu.showBicycleStands',
  checkedValue: false,
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <ThemeProvider theme={themes.SMTheme}>
      {children}
    </ThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<FormLabel />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<FormLabel {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<FormLabel {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual('Polkupyöräpysäköinti');
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<FormLabel {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toEqual('Polkupyöräpysäköinti');
  });
});
