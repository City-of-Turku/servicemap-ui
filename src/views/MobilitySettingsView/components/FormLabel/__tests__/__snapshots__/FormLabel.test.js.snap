// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLabel /> should work 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd injectIntl(FormLabel)-formLabel-1 injectIntl(FormLabel)-paddingMd-2 css-j204z7-MuiFormControlLabel-root"
  >
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
    >
      <span
        class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-18jccp8-MuiButtonBase-root-MuiSwitch-switchBase"
        role="switch"
      >
        <input
          aria-label="Polkupyöräpysäköinti (ei-runkolukittava)"
          class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    <p
      aria-label="Polkupyöräpysäköinti (ei-runkolukittava)"
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Polkupyöräpysäköinti (ei-runkolukittava)
    </p>
  </label>
</div>
`;
